### DISCUSSION TITLE
Lifecycle Hooks, Triggering JS events

### DISCUSSION THREAD

**Discussion by kundankarna1994** [livewire/livewire#6124] [7. Bugs (archived)]
### Livewire version

^3.0@beta

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/kundankarna1994/livewire-v3-testing

### Steps To Reproduce

1. composer install
2. Seed Database Using db:seed ( For now post only)
3. npm install ( for installing tailwind and daisyui)
4. npm run dev 

First , lifecycle hooks mentioned in the documents  
1. rendering
2. rendered
3. hydrate
were not triggering .

Also for editor i wanted to add CKEditor.

[CKEditor using vite](https://ckeditor.com/docs/ckeditor5/latest/installation/advanced/alternative-setups/integrating-from-source-vite.html)

Successfully added the editor then to integrate it i have to add this js code where i want to find all the elements with editor class and add CKEditor to it.
```
const elements = document.querySelectorAll(".editor");

elements.forEach((element) => {
    ClassicEditor.create(element)
    .then((editor) => {
        console.log(editor);
    })
    .catch((error) => {
        console.log(error);
    });
})
```

Okay All good for now, then i load the page to create post from the browser url. We can see a editor being loaded in place of textarea. But when i use wire navigate between components. JS which we have triggered at first wont trigger as we are only updating the html of the dom and the editor wont show.
Sorry if i have missed something. 

I tried to add the js code in rendered lifecycle but the method was not working.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@kundankarna1994 thanks for reporting! Can you trying initialising CKEditor using the `livewire:navigated` event and let us know how it goes? See here https://livewire.laravel.com/docs/navigate#dont-rely-on-domcontentloaded

Hope this helps!

**Reply by kundankarna1994** [reply]
@joshhanley Thank you for your help. Yep its working when i navigate. But not on component update.


If the components update like validation error or other . The components are again refetched and added to the dom. In this case again the editor wont initialize.

I have initialized the editor twice in my code 
1. When the DOM is loaded i have to initialize the editor. ( For the first page load)
2. Again when wire:navigate initialize again. ( For the wire navigate)

```function initializeEditor(){
    const elements = document.querySelectorAll(".editor");
    elements.forEach((element) => {
        ClassicEditor.create(element)
            .then((editor) => {
                console.log(editor);
            })
            .catch((error) => {
                console.log(error);
            });
    })
}

document.addEventListener('livewire:navigated', () => {
    initializeEditor() // initialize when wire navigated;
})

initializeEditor(); //initialize when DOM loaded```
